src/curves.rs

```diff
@@ -282,58 +525,235 @@ mod tests {
+
+        let brest_to_nancy = SphericalLineStringCurve::new(line_string![(x: BREST_LON, y: BREST_LAT), (x: NANCY_LON, y: NANCY_LAT)], 1.);
+        assert_relative_eq!(785636.8730262491, brest_to_nancy.length()); // 785636.8730262491 using [`HaversineLength`] and 787994.4363866252 using [`GeodesicLength`]
+    }
+
+    #[test]
+    fn spherical_is_valid() {
+        // Valid curve
+        let curve = SphericalLineStringCurve::new(line_string![(x: 0., y: 0.), (x: 1., y: 1.)], 1.);
+        assert!(curve.is_valid());
+
+        // Invalid curve: too few coordinates
+        let curve = SphericalLineStringCurve::new(line_string![(x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+
+        // Invalid curve: closed LineString with only 2 coordinates
+        let curve = SphericalLineStringCurve::new(line_string![(x: 0., y: 0.), (x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+
+        // Invalid curve: longitude > 180.
+        let curve = SphericalLineStringCurve::new(line_string![(x: 180.1, y: 0.), (x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+
+        // Invalid curve: longitude < -180.
+        let curve = SphericalLineStringCurve::new(line_string![(x: -180.1, y: 0.), (x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+
+        // Invalid curve: latitude > 90.
+        let curve = SphericalLineStringCurve::new(line_string![(x: 0., y: 90.1), (x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+
+        // Invalid curve: latitude > 180.
+        let curve = SphericalLineStringCurve::new(line_string![(x: 0., y: -90.1), (x: 0., y: 0.)], 1.);
+        assert!(!curve.is_valid());
+    }
+
+    #[test]
+    fn spherical_projection() {
+        let mut paris_to_new_york = SphericalLineStringCurve::new(line_string![(x: PARIS_LON, y: PARIS_LAT), (x: NEW_YORK_LON, y: NEW_YORK_LAT)], 1.);
+
+        // Point is located on the right (north) of the curve
+        let projected = paris_to_new_york.project(point! {x: -6.705403880820967, y: 51.42135181702875}).unwrap();
+        assert_eq!(701924.3809693493, projected.distance_along_curve);
+        assert_eq!(-67157.93913531031, projected.offset);
+
+        // Point is located on the left (south) of the curve
+        let projected = paris_to_new_york.project(point! {x: -12.250890759346419, y: 45.857650969554356}).unwrap();
+        assert_eq!(963365.3768036617, projected.distance_along_curve);
+        assert_eq!(625592.3211438804, projected.offset);
+
+        // Same point, but with an offset from the curve
+        paris_to_new_york.start_offset = 1000000.;
+        let projected = paris_to_new_york.project(point! {x: -12.250890759346419, y: 45.857650969554356}).unwrap();
+        assert_eq!(1963365.3768036617, projected.distance_along_curve);
+        assert_eq!(625592.3211438804, projected.offset);
+
+        // ################################################################################
```

Comment:
The tests seem to be of the same kind, what is this comment supposed to separate? maybe delete or replace with an explicit comment